### PR TITLE
Feature/orca 461 DB Connection Info Validation

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Added consistent validation of database connection parameters that may be used in SQL construction.

Addresses [ORCA-461: Update parameterization in db_deploy sql](https://bugs.earthdata.nasa.gov/browse/ORCA-461)

## Changes

* Switched db_deploy to updated shared_db code
* Added validation to several db parameters

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Deployed to AWS

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] Testing documentation
    - [x] Development documentation
    - [x] User documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

**Review by rizbihassan — APPROVED** [review]
looks good to me

**Review by bhazuka — CHANGES_REQUESTED** [review]
Overall looks good to me, A couple of minor items to look at an comment on.

File: shared_libraries/orca_shared/database/use_cases/validation.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def _validate_password(password: str, context: str, logger: logging.Logger) -> None:
+    """
+    Validates the given password against Orca restrictions.
+
+    Raises:
+        Exception: If value is empty or less than 12 characters.
+    """
+    if password is None or len(password) < 12:
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply shared_libraries/orca_shared/database/use_cases/validation.py:26]: https://bugs.earthdata.nasa.gov/browse/ORCA-573

File: shared_libraries/orca_shared/database/use_cases/validation.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        msg = f"{context} password must be at least 12 characters long."
+        logger.critical(msg)
+        raise Exception(msg)
+
+
+def validate_postgres_name(name: str, context: str, logger: logging.Logger) -> None:
+    """
+    Validates the given name against documented Postgres restrictions.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply shared_libraries/orca_shared/database/use_cases/validation.py:34]: Added some newly supported characters. https://bugs.earthdata.nasa.gov/browse/ORCA-572 for remainder.

**Pull request merged**